src/gluonts/model/estimator.py

```diff
@@ -69,6 +69,21 @@ def train(
     def from_hyperparameters(cls, **hyperparameters):
         return from_hyperparameters(cls, **hyperparameters)
 
+    @classmethod
+    def derive_auto_fields(cls, train_iter):
+        return {}
+
+    @classmethod
+    def from_inputs(cls, train_iter, **params):
+        # auto_params usually include `use_feat_dynamic_real`, `use_feat_static_cat` and `cardinality`
+        auto_params = cls.derive_auto_fields(train_iter)
+        # FIXME: probably params should take precedence over auto_params, since they were deliberately set,
+        #   however, on that case this method does not make sense, since if params says `use_feat_dynamic_real`=True
+        #   but `auto_params`=False, then this will lead to an error, since the appropriate data does not exist.
```

Comment:
I think this is fine, if the user sets incorrect parameters, there is not much we can do.

Reply:
Well, we could still set the Warning?

Reply:
Or rather in this case I think the program will fail and say that a parameter has been specified twice, so yeah, we don't need to warn, you're right.